src/app/blocks/load-schema-block/load-schema-block.component.ts

```diff
@@ -4,6 +4,157 @@ import { get, has, clone } from 'lodash-es';
+
+        }
+      }
+    };
+
+    return newObject; // Return a copy of the original object with all keys converted and replaced by their default values where applicable!
+  }
+
+  function arrayOfUUIDsToDict(object) {
+    function arrayChildrenAllHaveUUIDs(array) {
+      for (var i = 0; i < array.length; i++) {
+        if (!array[i]["uuid"]) return false; // If any child object does not have a UUID, we return false and do nothing to this array.  
+      }
+
+      return true; // Otherwise we can assume that all children of the array have UUIDs, so it is safe to convert them into an object with those keys!
+
+    }
+
+    var newObject = clone(object); // Make a copy of the original object.  
+    for (var key in newObject) {// Loop through each key in the original object:        
+      if (typeof newObject[key] === "object") { // If it is an array or an object, recurse into it: 
+        newObject[key] = arrayOfUUIDsToDict(newObject[key]); // Recursively call this function on any objects found within this one, replacing them with their converted versions as they are returned from recursion.  
+
+        if (Array.isArray(newObject[key]) && arrayChildrenAllHaveUUIDs(newObject[key])) {
+          var tempObj = {};
+          for (var i = 0; i < newObject[key].length; i++) { // Loop through the original child objects in the original parent object's key value:  
+            tempObj[newObject[key][i]["uuid"]] = newObject[key][i]; // Add each child object to a temporary object, using the UUID as the key.  
+            delete tempObj[newObject[key][i]["uuid"]].uuid; // Delete the UUID from each child object once it is used as a key instead.  
+          }
+
+          newObject[key] = tempObj; // Replace the original parent object's key value with our temporary object containing all of the child objects, now using their UUIDs as keys.  
+        }
+      }
+    };
+
+    return newObject; // Return a copy of the original object with all arrays converted to objects where applicable!
+  }
+
+
+  function convertToValidJSONSchema(template) {
+    let schema = {
+      "type": "object",
+      "properties": {}
+    }
+    for (let key in template) {
+      if (typeof template[key] === "object") {
+        // Only modify invalid JSON schema fragments
+        if (template[key].type === undefined) {
+          schema.properties[key] = convertToValidJSONSchema(template[key])
+        } else {
+          schema.properties[key] = template[key]
+        }
+      } else {
+        schema.properties[key] = {
+          "type": "string",
+          "default": template[key],
+          "readOnly": true,
+          "title": template[key]
+        }
+      }
+    }
+    return schema
+  }
+  console.log("returning test")
+  //return blockTypeDefaults["c5bfac02-f0b6-4c31-8fdf-18de02667ee9"];
+  return convertToValidJSONSchema(arrayOfUUIDsToDict(applyBlockTypeDefaults(convertKeys(template))))
+}
+
+
```

Comment:
This probably needs some introduction or overview. It's a chunk of new tech that may be core to future work. I'd like to little intro to what is going on to that I understand the higher level concepts, before needing to get into the details of the code.

Reply:
You should find it better commented now.